hal/ubuntu/qemu.sh

```
@@ -0,0 +1,598 @@
+  QEMU_OPTS="$QEMU_OPTS -drive file=$CUSTOM_IMAGE,if=none,id=disk0,format=qcow2"
+  QEMU_OPTS="$QEMU_OPTS -device virtio-scsi-pci,id=scsi,disable-legacy=on"
+  QEMU_OPTS="$QEMU_OPTS -device scsi-hd,drive=disk0"
+
+  if [ "$TDX_AVAILABLE" = true ]; then
+    echo "Starting QEMU VM with Intel TDX (Confidential VM)..."
+    QEMU_OPTS=$(echo "$QEMU_OPTS" | sed "s/-name $VM_NAME/-name $VM_NAME,process=$VM_NAME,debug-threads=on/")
+    QEMU_OPTS=$(echo "$QEMU_OPTS" | sed "s/-m $RAM//")
+    QEMU_OPTS="$QEMU_OPTS -object memory-backend-memfd,id=ram1,size=$RAM,share=true,prealloc=false"
+    QEMU_OPTS="$QEMU_OPTS -m $RAM"
+    QEMU_OPTS="$QEMU_OPTS -cpu host,pmu=off"
+    QEMU_OPTS="$QEMU_OPTS -object {\"qom-type\":\"tdx-guest\",\"id\":\"tdx0\",\"quote-generation-socket\":{\"type\":\"vsock\",\"cid\":\"2\",\"port\":\"4050\"}}"
```

Shellcheck (0.11.0)

[warning] Quotes/backslashes will be treated literally. Use an array.

(SC2089)